File: app/src/main/java/com/ranzan/mojopizzaclone/DataActivity.java
package com.ranzan.mojopizzaclone;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.ranzan.mojopizzaclone.Menu.All_in_oneFragment;
import com.ranzan.mojopizzaclone.Menu.Detailed_Item_Fragment;
import com.ranzan.mojopizzaclone.communication.FragmentListener;

public class DataActivity extends AppCompatActivity implements FragmentListener {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_data);
        Intent intent = getIntent();
        int n = intent.getIntExtra("pos", 0);
        Bundle bundle=new Bundle();
        bundle.putInt("pos",n);
        All_in_oneFragment fragment = new All_in_oneFragment();
        fragment.setFragmentListener(this);
        fragment.setArguments(bundle);
        getSupportFragmentManager().beginTransaction().replace(R.id.contaner, fragment).commit();
    }


    @Override
    public void fragmentListener(Bundle bundle) {
        Detailed_Item_Fragment fragment = new Detailed_Item_Fragment();
        fragment.setArguments(bundle);
        getSupportFragmentManager().beginTransaction().add(R.id.contaner, fragment).addToBackStack("").commit();
    }
}

File: app/src/main/java/com/ranzan/mojopizzaclone/DataActivity.kt
package com.ranzan.mojopizzaclone

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import com.ranzan.mojopizzaclone.Menu.All_in_oneFragment
import com.ranzan.mojopizzaclone.Menu.Detil_ItemFragment
import com.ranzan.mojopizzaclone.communication.FragmentListener

class DataActivity : AppCompatActivity(), FragmentListener {
    private var fragmentManager: FragmentManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_data)
        val intent = intent
        val n = intent.getIntExtra("pos", 0)
        val bundle = Bundle()
        bundle.putInt("pos", n)
        val fragment = All_in_oneFragment()
        fragment.setFragmentListener(this)
        fragment.arguments = bundle
        fragmentManager = supportFragmentManager
        fragmentManager!!.beginTransaction().add(R.id.contaner, fragment, bundle.toString())
            .commit()
    }

    override fun fragmentListener(bundle: Bundle?) {
        val fragmentTransaction = fragmentManager!!.beginTransaction()
        val fragment = Detil_ItemFragment()
        fragment.arguments = bundle
        fragmentTransaction.add(R.id.contaner, fragment).addToBackStack("").commit()
    }
}